### PR TITLE
Update training pipelines to pytorch 1.10

### PR DESCRIPTION
**Pull request by raviskolli**
Added Pytorch 1.10 support for training python packages and pipelines

### CODE REVIEW

**Comment by snnn** [comment]
Please wait a moment.  Currently we have some issues on the existing pipelines.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+stages:
+- template: templates/py-packaging-training-cuda-stage.yml
+  parameters:
+    build_py_parameters: --enable_training --update --build
+    torch_version: '1.10.0'
+    cuda_version: '10.2'
+    gcc_version: 10
+    cmake_cuda_architectures: 37;50;52;60;61;70;75;80;86
+    docker_file: Dockerfile.manylinux2014_training_cuda11_3
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml:11]: The file name is cuda102 but the actual cuda version is 11_3?

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch182-cuda102.yml [hunk 2]
```diff
@@ -1,12 +0,0 @@
-trigger: none
-
-stages:
-- template: templates/py-packaging-training-cuda-stage.yml
-  parameters:
-    build_py_parameters: --enable_training --update --build
-    torch_version: '1.8.2'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch182-cuda102.yml:7]: Should the old docker files be deleted too?

**Comment by snnn** [comment]
The pipelines are good now except AMD GPU pipelines. But you won't touch AMD GPU pipelines. So, please go ahead.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml (hunk 1 above)
Comment (raviskolli) [reply tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml:11]: Updated to use cuda 10_2 dockerfile.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch182-cuda102.yml (hunk 2 above)
Comment (raviskolli) [reply tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch182-cuda102.yml:7]: Yes, I believe I've deleted all the old dockerfiles. Let me know if I missed any.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml [hunk 3]
```diff
@@ -4,9 +4,10 @@ stages:
 - template: templates/py-packaging-training-cuda-stage.yml
   parameters:
     build_py_parameters: --enable_training --update --build
-    torch_version: '1.8.2'
+    torch_version: '1.10.0'
     cuda_version: '10.2'
-    gcc_version: 8
-    cmake_cuda_architectures: 35;37;50;52;60;61;70
+    gcc_version: 10
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml:9]: GCC 10 doesn't work with CUDA 10.2

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3 [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+
+FROM build_base AS build_swig
+COPY build_scripts/build-swig.sh /build_scripts/
+RUN export SWIG_ROOT=swig-4.0.2 && \
+    export SWIG_HASH=d53be9730d8d58a16bf0cbd1f8ac0c0c3e1090573168bfa151b01eb47fa906fc && \
+    export SWIG_DOWNLOAD_URL=https://sourceforge.net/projects/swig/files/swig/${SWIG_ROOT} && \
+    export PCRE_ROOT=pcre-8.44 && \
+    export PCRE_HASH=aecafd4af3bd0f3935721af77b889d9024b2e01d96b58471bd91a3063fb47728 && \
+    export PCRE_DOWNLOAD_URL=https://ftp.pcre.org/pub/pcre && \
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3:97]: This url doesn't work any more. I just updated the other docker files, please check them.

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml (hunk 3 above)
Comment (raviskolli) [reply tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1100-cuda102.yml]: Downgraded gcc to 8

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3 (hunk 4 above)
Comment (raviskolli) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3]: Updated to the latest url

**Review by snnn — APPROVED** [review]
LGTM.

**Comment by baijumeswani** [comment]
May we also update the CI pipeline file: https://github.com/microsoft/onnxruntime/blob/2beb873c6bbf174e66f072d88ebd765bb94c6ff1/tools/ci_build/github/azure-pipelines/templates/orttraining-linux-gpu-ortmodule-test-ci-pipeline.yml#L36

**Pull request merged**